File: 프로그래머스/2/87946. 피로도/피로도.py
from itertools import permutations
def solution(k, dungeons):
    result = 0
    
    for index_arr in permutations([i for i in range(len(dungeons))]):
        tmp_k = k 
        go = 0
        for idx in index_arr: 
            if tmp_k >= dungeons[idx][0]:
                tmp_k -= dungeons[idx][1]
                go += 1
        result = max(go, result)
        
    return result

File: 프로그래머스/2/87946. 피로도/피로도.js
function solution(k, dungeons) {
    let answer = 0
    let visited = new Array(dungeons.length).fill(false)
    
    const DFS = (hp , count) => {
        for(let i = 0 ; i < dungeons.length ; i++) {
            if(!visited[i] && hp >= dungeons[i][0]){
                visited[i] = true
                DFS(hp - dungeons[i][1] , count + 1)
                visited[i] = false
            }    
        }
        answer = Math.max(answer, count)
    }
    
    DFS(k,0)
    return answer
}